src/plugin/pid/pid.cpp

```diff
@@ -36,11 +36,12 @@ using namespace dmtcp;
 extern "C" pid_t dmtcp_update_ppid();
 
 static string pidMapFile;
-map<pthread_mutex_t*, pid_t>& mapMutexVirtTid()
+dmtcp::map<pthread_mutex_t*, pid_t>& mapMutexVirtTid()
 {
-  static map<pthread_mutex_t*, pid_t> *instance = NULL;
+  static dmtcp::map<pthread_mutex_t*, pid_t> *instance = NULL;
   if (instance == NULL) {
-    instance = new map<pthread_mutex_t*, pid_t>();
+    void *buffer = JALLOC_MALLOC(1024*1024);
+    instance = new (buffer) dmtcp::map<pthread_mutex_t*, pid_t>();
```

Comment:
Is this equal to
`instance = reinterpret_cast<dmtcp::map<pthread_mutex_t*, pid_t> *> (buffer)` ?
I'm no C++ expert, and that's why it takes me a while to understand what it does. @karya0 should be the judge here.